smartsim/_core/entrypoints/dragon.py

```diff
@@ -0,0 +1,178 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2023 Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import json
+import os
+import signal
+import socket
+import textwrap
+import typing as t
+from types import FrameType
+
+import zmq
+
+import smartsim._core.utils.helpers as _helpers
+from smartsim._core.launcher.dragon.dragonBackend import DragonBackend
+from smartsim._core.schemas import (
+    DragonBootstrapRequest,
+    DragonBootstrapResponse,
+    DragonShutdownResponse,
+)
+from smartsim._core.schemas.dragonRequests import request_serializer
+from smartsim._core.schemas.dragonResponses import response_serializer
+from smartsim._core.utils.network import get_best_interface_and_address
+
+# kill is not catchable
+SIGNALS = [signal.SIGINT, signal.SIGQUIT, signal.SIGTERM, signal.SIGABRT]
+
+SHUTDOWN_INITIATED = False
+
+
+def handle_signal(signo: int, _frame: t.Optional[FrameType]) -> None:
+    if not signo:
+        print("Received signal with no signo")
+    else:
+        print(f"Received {signo}")
+    cleanup()
+
+
+context = zmq.Context()
+
+"""
+Dragon server entrypoint script
+"""
+
+DBPID: t.Optional[int] = None
+
+
+def print_summary(network_interface: str, ip_address: str) -> None:
+    zmq_config = {"interface": network_interface, "address": ip_address}
+
+    with open("dragon_config.log", "w", encoding="utf-8") as dragon_config_log:
```

Comment:
Can we pull this file name out and put it in `smartsim._core.config.Config` in case something else needs to touch it (and we want to avoid potentially changing the name in multiple places)

Reply:
Upon review, it looks like we already pulled the filename int `Config` but just missed updating this reference.